source/uuid.rst

```diff
@@ -0,0 +1,329 @@
+   
+      /*
+       * Decode a subtype 3 or 4 to a UUID, according to the UUID    
+       * representation, erroring when subtype does not match the
+       * representation.
+       */
+      Uuid asUuid(UuidRepresentation representation)
+   }
+
+Implicit decoding and encoding
+------------------------------
+
+A new driver for a language with a native UUID type MUST NOT implicitly encode from or decode to the native UUID type.  Rather, explicit conversion MUST be used as described in the previous section.
+
+Drivers that already do such implicit encoding and decoding SHOULD support a URI option, uuidRepresentation, which controls the default behavior of the UUID codec. Alternatively, a driver MAY specify the UUID representation via global state. 
+
+
+
+.. list-table::
+   :header-rows: 1
+   :widths: 1 1 3 1 1
+
+   * - Value
+     - Default?
+     - Encode to
+     - Decode subtype 4 to
+     - Decode subtype 3 to
+     
+   * - unspecified
+     - yes
+     - raise error 
+     - BsonBinary
+     - BsonBinary
+
+   * - standard
+     - no
+     - BSON binary subtype 4 
+     - native UUID
+     - BsonBinary
+
+   * - csharpLegacy
+     - no
+     - BSON binary subtype 3 with C# legacy byte order 
+     - BsonBinary
+     - native UUID
+
+   * - javaLegacy
+     - no
+     - BSON binary subtype 3 with Java legacy byte order 
+     - BsonBinary
+     - native UUID
+
+   * - pythonLegacy
+     - no
+     - BSON binary subtype 3 with standard byte order 
+     - BsonBinary
+     - native UUID
+
+For scenarios where the application makes the choice (e.g. a POJO with a field of type UUID), or when serializers are strongly typed and are constrained to always return values of a certain type, the driver will raise an exception in cases where otherwise it would be required to decode to a different type (e.g. BsonBinary instead of UUID or vice versa).
+
+Note also that none of the above applies when decoding to strictly typed maps, e.g. a Map<String, BsonValue> like Java or .NET's BsonDocument class.  In those cases the driver is always decoding to BsonBinary, and applications would use the asUuid methods to explicitly convert from BsonBinary to UUID.
+
+
+Implementation Notes
+--------------------
+
+Since changing the default UUID representation can reasonably be considered a backwards-breaking change, drivers that implement the full specification should stage implementation according to semantic versioning guidelines.  Specifically, support for this specification can be added to a minor release, but with several exceptions: 
+
+The default UUID representation should be left as is (e.g. JAVA_LEGACY for the Java driver) rather than be changed to UNSPECIFIED.  In a subsequent major release, the default UUID representation can be changed to UNSPECIFIED (along with appropriate documentation indicating the backwards-breaking change).
+Subtype 4 BSON Binary values should continue to be decoded to native UUID values regardless of UUID representation
```

Comment:
This line seems to contradict what the table above (under "Implicit decoding and encoding") says.

Reply:
Which line of the table?  I don't see the conflict.

Reply:
The table says to decode subtype 4 to either a BsonBinary or a UUID depending on which line of the table you look at.

This statement says that subtype 4 BsonBinary values should continue to be decoded to native UUID values regardless of the UUID representation.

Which is it?

Reply:
I see now.  I think this implementation note is incorrect and should be removed.  Any objections?

Reply:
Agreed. Should we add a line that says "Drivers should document that the default UUID representation will change to UNSPECIFIED in the next major release?"

Reply:
Done.  I don't love a MUST in an implementation note, but there isn't a good place to put it other than this.